scripts/generate-app-config/generateRouteConfig.js

```diff
@@ -0,0 +1,205 @@
+    if (page.content) {
+      content[routePath] = contentRouteItem(routePath, { contentPath: page.content }, page.props, page.type, routeImporter);
+    } else if (redirectRoute) {
+      // If a redirect Route has been identified, redirect to it.
+      content[routePath] = contentRouteItem(routePath, { contentPath: Redirect, name: '{ Redirect }', identifier: 'Redirect' }, { to: redirectRoute }, 'js', routeImporter);
+    } else {
+      content[routePath] = contentRouteItem(routePath, { contentPath: placeholder.content, name: 'TerraDevSitePlaceholder' }, placeholder.props, 'js', routeImporter);
+    }
+
+    return { content, menu, menuItems };
+  }, { content: {}, menu: {} })
+);
+
+/**
+* Create a page config object for the top level page.
+*/
+const getPageConfig = (name, pagePath, pages, type, siteConfig, routeImporter) => {
+  const { readMeContent } = siteConfig;
+  const config = {
+    name: startCase(name),
+    path: pagePath,
+    pages,
+  };
+
+  // Special logic to add a home component with a readme if readme content is provided in site config and no other home items are found.
+  if (type === 'home' && readMeContent) {
+    config.content = TerraDocTemplate;
+    config.props = { readme: routeImporter.addImport(ImportAggregator.relativePath(readMeContent)) };
+  }
+
+  return config;
+};
+
+/**
+* Build out the page config for the top level link.
+*/
+const getLinkRoute = (link, pageConfig, siteConfig, routeImporter) => {
+  let pages = [];
+  let type;
+
+  // If a link has more than one type, we'll have to build out at least three page configs.
+  if (link.pageTypes.length > 1) {
+    // Do no sort link types. It's up to the consumer to order that array.
+    pages = link.pageTypes.reduce((acc, pageType) => {
+      acc.push(getPageConfig(pageType, `/${kebabCase(pageType)}`, pageConfig[pageType], pageType, siteConfig, routeImporter));
+      return acc;
+    }, []);
+  } else {
+    // If there is only one type, just grab the first one.
+    pages = pageConfig[link.pageTypes[0]];
+    type = link.pageTypes[0];
+  }
+
+  return [getPageConfig(link.text, link.path, pages, type, siteConfig, routeImporter)];
+};
+
+/**
+* Build out route configuration and modify the navigation items config too.
+*/
+const routeConfiguration = (siteConfig, pageConfig) => {
+  if (!pageConfig) {
+    return undefined;
+  }
+  const routeImporter = new ImportAggregator();
+  const { placeholderSrc } = siteConfig;
+  const navConfig = siteConfig.navConfig;
+  const navigation = navConfig.navigation;
+  const validLinks = navigation.links ? navigation.links.filter(link => link.path && link.text && link.pageTypes) : [];
+
+  // Setup the placeholder object.
+  const placeholder = { content: PlaceholderPath, props: { src: placeholderSrc } };
+
+  // Spin through the valid links to build out the route config.
+  const config = validLinks.reduce((acc, link) => {
+    let content = acc.content;
```

Comment:
Did you add content and menu as navigation.link options? if not, we should just assign these as `{}` I think

Reply:
This is on purpose, content and menu are continually added to as we reduce